.docker/qgis.dockerfile

```diff
@@ -42,4 +45,28 @@ RUN cmake \
  && ninja install \
  && rm -rf /usr/src/QGIS
 
+################################################################################
+# Python testing environment setup
+
+# Add QGIS test runner
+COPY .docker/qgis_resources/test_runner/qgis_* /usr/bin/
+
+# Make all scripts executable
+RUN chmod +x /usr/bin/qgis_*
+
+# Add supervisor service configuration script
+COPY .docker/qgis_resources/supervisor/supervisord.conf /etc/supervisor/
+COPY .docker/qgis_resources/supervisor/supervisor.xvfb.conf /etc/supervisor/supervisor.d/
+
+# Python paths are for
+# - kartoza images (compiled)
+# - deb installed
+# - built from git
+# needed to find PyQt wrapper provided by QGIS
+ENV PYTHONPATH=/usr/share/qgis/python/:/usr/lib/python3/dist-packages/qgis:/usr/share/qgis/python/qgis
+
+
 WORKDIR /
+
+# Run supervisor
+CMD ["/usr/bin/supervisord", "-c", "/etc/supervisor/supervisord.conf"]
```

Comment:
does it make sense to have this explicitly written here or it could done in a dedicated script whenever doing testing for plugins?
It seems a bit strange to have this by default no?

Reply:
Well, there are many ways to do that, I didn't find it strange because the purpose of the original image was to run headless tests and xvfb was required to run them, so launching it in the background served the double purpose of launching a required service and keep the docker up (it would just exit immediately without a CMD).
In any event I would prefer to not change this behavior because it would require adaptation of all the existing CI configurations and local scripts, which expect the containers to stay up after a simmple `docker run -d --name qgis-testing-environment -v ${TRAVIS_BUILD_DIR}:/tests_directory -e DISPLAY=:99 qgis-testing-environment` command.